stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicPublish.java

```diff
@@ -60,8 +61,8 @@ public static Object basicPublish(Strand strand, ObjectValue channelObjectValue,
         try {
             ChannelUtils.basicPublish(channel, routingKey, messageContent.toString().getBytes(StandardCharsets.UTF_8),
                     defaultExchangeName, properties);
-            if (transactionContext != null) {
-                transactionContext.handleTransactionBlock();
+            if (isInTransaction && transactionContext != null) {
+                transactionContext.handleTransactionBlock(strand);
             }
```

Comment:
`transactionContext` cannot be null, because we are creating at channel creation. Shall we stop publishing the message and give an error if any case it is null and `isInTransaction`.